subsys/mgmt/mcumgr/smp.c

```diff
@@ -54,19 +54,6 @@ zephyr_smp_alloc_rsp(const void *req, void *arg)
 	return rsp_nb;
 }
 
-static void
-zephyr_smp_trim_front(void *buf, size_t len, void *arg)
-{
-	struct net_buf *nb;
-
-	nb = buf;
-	if (len > nb->len) {
```

Comment:
I ques that you know what you are doing.
This check is droped.

Reply:
Line 99 of the `zephyr_smp_split_frag`, where the trim used to be called checks this before calling trim.